src/azul/service/elasticsearch_service.py

```diff
@@ -328,35 +328,45 @@ def _generate_paging_dict(self, es_response, pagination):
         # ...else use search_after/search_before pagination
         es_hits = es_response['hits']['hits']
         count = len(es_hits)
+        self_url = pagination['self_url'] + f"?size={pagination['size']}"
         if 'search_before' in pagination:
             # hits are reverse sorted
             if count > pagination['size']:
                 # There is an extra hit, indicating a previous page.
                 count = count - 1
                 search_before = es_hits[count - 1]['sort']
+                params = f'&search_before={search_before[0]}&search_before_uid={search_before[1].replace("#", "%23")}'
```

Comment:
Don't manually encode. There are built-ins for that:

https://docs.python.org/3.6/library/urllib.parse.html#urllib.parse.urlencode